libarchive/archive_write_set_format_pax.c

```diff
@@ -1117,15 +1119,17 @@ archive_write_pax_header(struct archive_write *a,
 
 	/*
 	 * Technically, the mtime field in the ustar header can
-	 * support 33 bits, but many platforms use signed 32-bit time
-	 * values.  The cutoff of 0x7fffffff here is a compromise.
+	 * support 33 bits. We are using all of them to keep
+	 * tar/test/test_option_C_mtree.c simple and passing after 2038.
+	 * Platforms that use signed 32-bit time values need to fix
+	 * their handling of timestamps anyway.
```

Comment:
I suggest moving the above paragraph to go with the definition of `USTAR_MAX_MTIME` since this is the justification for that constant.  The paragraph below should stay here; it's about the check and structure of this part of the code.

Reply:
I moved the comment up.